spk/dnscrypt-proxy/src/service-setup.sh

```diff
@@ -29,44 +32,56 @@ pgrep () {
     ps -w | grep "[^]]$1" >> "${LOG_FILE}" 2>&1
 }
 
-disable_dhcpd_dns_port () {
-    if [ "$1" == "no" ] && [ -f /etc/dhcpd/dhcpd-custom-custom.conf ]; then
-        echo "Port 0 - dhcpd (dnsmasq) enabled: $1" >> "${LOG_FILE}"
-        echo "enable=\"$1\"" > /etc/dhcpd/dhcpd-custom-custom.info
+forward_dns_dhcpd () {
+        echo "dns forwarding - dhcpd (dnsmasq) enabled: $1" >> "${LOG_FILE}"
         /etc/rc.network nat-restart-dhcp >> "${LOG_FILE}" 2>&1
-    elif [ "$1" == "yes" ] && netstat -na | grep ":53 " >> "${LOG_FILE}" 2>&1; then
-        echo "Port 53 is in use" >> "${LOG_FILE}"
+    elif [ "$1" == "yes" ]; then
         if pgrep "dhcpd.conf"; then  # if dhcpd (dnsmasq) is enabled and running
-            echo "Port 0 - dhcpd (dnsmasq) enabled: $1" >> "${LOG_FILE}"
-            echo "port=0" > /etc/dhcpd/dhcpd-custom-custom.conf
-            echo "enable=\"$1\"" > /etc/dhcpd/dhcpd-custom-custom.info
+            echo "dns forwarding - dhcpd (dnsmasq) enabled: $1" >> "${LOG_FILE}"
+            echo "server=127.0.0.1#${BACKUP_PORT}" > /etc/dhcpd/dhcpd-dnscrypt-dnscrypt.conf
+            echo "enable=\"$1\"" > /etc/dhcpd/dhcpd-dnscrypt-dnscrypt.info
             /etc/rc.network nat-restart-dhcp >> "${LOG_FILE}" 2>&1
         else
             echo "pgrep: no process with 'dhcpd.conf' found" >> "${LOG_FILE}"
         fi
-    else
-        echo "Port 53 is free" >> "${LOG_FILE}"
-        rm -f /etc/dhcpd/dhcpd-custom-custom.conf
     fi
 }
 
 service_prestart () {
-    disable_dhcpd_dns_port "yes"
+    echo "service_preinst ${SYNOPKG_PKG_STATUS}" >> "${INST_LOG}"
+
+    # This fixes https://github.com/SynoCommunity/spksrc/issues/3468
+    # This can't be done at install time. see:
+    #  https://github.com/SynoCommunity/spksrc/blob/e914a32600e65f80131ae09913f1b6f6a2dd8b13/mk/spksrc.service.installer#L307-L319
+    chown root:root "${SYNOPKG_PKGDEST}/ui/index.cgi"
```

Comment:
@ymartin59 Yes I'm happy as long as you are ok with the chroot here ^

Reply:
Sorry for the delay, I've been working on fixing bugs. The CGI program uses [/usr/syno/synoman/webman/modules/authenticate.cgi](https://originhelp.synology.com/developer-guide/integrate_dsm/web_authentication.html) to check if the user is logged-in and `/usr/syno/synoman/webman/initdata.cgi` to check that the user is an admin (only on the SRM because I have not found `/initdata.cgi` on the DSM). As far as I can tell root is required to execute those CGI programs. On the SRM the permissions given here are the same as from the web server user (system). The DSM executes CGI programs differently but I haven't had the time to test all of the permission gotchas/differences between the 2 systems. Having root makes this easier. I think there must be a better way, but I havn't found it yet.

Reply:
Oh and `/usr/syno/synoman/webman/login.cgi` is also used for the CSRF tokens.